src/org/thoughtcrime/securesms/database/loaders/RecentPhotosLoader.java

```diff
@@ -36,7 +36,7 @@ public RecentPhotosLoader(Context context) {
 
   @Override
   public Cursor loadInBackground() {
-    if (Permissions.hasAll(context, Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+    if (Permissions.hasAll(context, Manifest.permission.READ_EXTERNAL_STORAGE)) {
```

Comment:
READ_EXTERNAL_STORAGE sounds reasonable, however, we used WRITE_EXTERNAL_STORAGE before - and also Signal still uses WRITE_EXTERNAL_STORAGE (Signals RecentPhotosLoader.java is quite the same).

so, i do not really get that part (/me currently playing around with the android6 phone and trying to get the preview rail back there again)

Reply:
It gets even weirder, what you are experiencing on android6 has a completely unrelated cause.

But also there, Signal should have the same issue.

Sometimes it would be nice to build Signal and poke around a bit (adding log statements) but it's not building out of the box and it's probably not worth the effort.